crypto/internal.h

```diff
@@ -1149,6 +1149,28 @@ OPENSSL_EXPORT int OPENSSL_vasprintf_internal(char **str, const char *format,
                                               va_list args, int system_malloc)
     OPENSSL_PRINTF_FORMAT_FUNC(2, 0);
 
+
+// Safety Macros
+
+// Inspired by s2n-tls
+
+// __AWS_LC_ENSURE checks if |cond| is true nothing happens, else |action| is called
+#define __AWS_LC_ENSURE(cond, action) \
```

Comment:
I ain't the super fan of these macro's. It quickly inflate into tens of macro's to capture different return types/values and different error behaviour (e.g. goto's).

mark as experimental?

Reply:
It's in an internal header so we shouldn't have to worry about external customers using it, but I'll mark the section as experimental in case they ever get moved and we forget.